zproject/urls.py

```diff
@@ -220,6 +218,12 @@
         {'PATCH': 'patch_bot_backend',
          'DELETE': 'deactivate_bot_backend'}),
 
+) + patterns('zerver.views.upload',
+    url(r'^user_uploads/(?P<realm_id_str>(\d*|unk))/(?P<filename>.*)', 'rest_dispatch',
+            {'GET': 'api_get_uploaded_file'}),
```

Comment:
Does it make sense to have an API route for serving files?  I think it could make more sense to not prefix the URIs for serving files with either /json or /api/v1, but to accept both API style and JSON style authentication on the routes to fetch user-uploaded files; the URLs would certainly be cleaner that way.  What do you think?